src/polyswarmclient/__init__.py

```
@@ -47,11 +47,11 @@ def is_valid_ipfs_uri(ipfs_uri):
     # TODO: Further multihash validation
     try:
         return len(ipfs_uri) < 100 and base58.b58decode(ipfs_uri)
-    except Exception:
-        pass
-
-    logger.error('Invalid IPFS URI: %s', ipfs_uri)
-    return False
+    except TypeError:
+        logger.error('Invalid IPFS URI: %s', ipfs_uri)
+        return False
+    except Exception as err:
+        logger.exception('Unexpected error: %s', err)
```

This case needs a `return False`, right?

---

Or better: `return False` outside of the exception clauses.

Now I see what you were asking about before.

---

@myhndl My idea here is that we should bubble up the exception because we've encountered a condition that is _more_ exceptional than a mangled `ipfs_uri`.

---

@myhndl Thank you! Changes made.